src/app/modules/main/wallet_section/module.nim

```diff
@@ -105,45 +110,59 @@ method delete*(self: Module) =
   self.sendModule.delete
   self.controller.delete
   self.view.delete
+
   if not self.addAccountModule.isNil:
     self.addAccountModule.delete
 
 method updateCurrency*(self: Module, currency: string) =
   self.controller.updateCurrency(currency)
 
-method switchAccount*(self: Module, accountIndex: int) =
-  self.assetsModule.switchAccount(accountIndex)
-  self.collectiblesModule.switchAccount(accountIndex)
-  self.transactionsModule.switchAccount(accountIndex)
-  self.overviewModule.switchAccount(accountIndex)
-
-method switchAccountByAddress*(self: Module, address: string) =
-  let accountIndex = self.controller.getIndex(address)
-  self.switchAccount(accountIndex)
-
 method setTotalCurrencyBalance*(self: Module) =
   self.view.setTotalCurrencyBalance(self.controller.getCurrencyBalance())
 
+method notifyFilterChanged(self: Module) =
+  self.overviewModule.filterChanged(self.filter.addresses, self.filter.chainIds)
+  self.assetsModule.filterChanged(self.filter.addresses, self.filter.chainIds)
+  self.collectiblesModule.filterChanged(self.filter.addresses, self.filter.chainIds)
+  self.transactionsModule.filterChanged(self.filter.addresses, self.filter.chainIds)
+
 method getCurrencyAmount*(self: Module, amount: float64, symbol: string): CurrencyAmount =
   return self.controller.getCurrencyAmount(amount, symbol)
 
+
+method setFilterAddress*(self: Module, address: string) =
+  self.filter.setAddress(address)
+  self.notifyFilterChanged()
+
 method load*(self: Module) =
   singletonInstance.engine.setRootContextProperty("walletSection", newQVariant(self.view))
 
+  self.events.on(SIGNAL_WALLET_ACCOUNT_UPDATED) do(e:Args):
+    self.notifyFilterChanged()
   self.events.on(SIGNAL_WALLET_ACCOUNT_SAVED) do(e:Args):
+    let args = AccountSaved(e)
     self.setTotalCurrencyBalance()
+    self.filter.setAddress(args.account.address)
+    self.notifyFilterChanged()
   self.events.on(SIGNAL_WALLET_ACCOUNT_DELETED) do(e:Args):
-    self.switchAccount(0)
+    let args = AccountDeleted(e)
     self.setTotalCurrencyBalance()
+    self.filter.removeAddress(args.address)
+    self.notifyFilterChanged()
```

Comment:
hmm wondering if this signal should be coming from the filter itself. For example, if an account is removed which was not active in the filter, whatever this signal does shouldn't need to be triggered.

Reply:
that's what i tried initialy but nim lang didn't help me there. I couldn't have interface working. And passing modules 1 by 1 to the filter didn't make sense

Reply:
hmm not sure if I explained myself correctly.

I meant simply having a signal `FilterChanged` which comes from the filter whenever any function (`updateNetworks`, `removeAddress`, `setAddress`) is called, since the filter is the only one that knows if something actually changed.
Then, this module connects to that signal and calls `notifyFilterChanged` like you're doing here, but in a single place.

Anyway, nitpicking, just seemed tidier.